File: rabbitmq/js/consumer.js
const amqp = require("amqplib")
connect();
async function connect() {
  try {
    const conn = await amqp.connect("amqp://localhost:5672");
    const channel = await conn.createChannel();
    const res = await channel.assertQueue("jobs");
    channel.consume("jobs", message => {
      console.log(message.content.toString())
      channel.ack(message);
    })
    console.log("waiting for messages...")
  } catch(err) {
    console.error(err);
  }
}

File: rabbitmq/go/consumer/consumer.go
package main

import (
	"log"

	"github.com/rabbitmq/amqp091-go"
)

func failOnError(err error, msg string) {
	if err != nil {
		log.Panicf("%s: %s", msg, err)
	}
}

func main() {
	conn, err := amqp091.Dial("amqp://user:password@localhost:5672/")
	failOnError(err, "failed to conenct to rabbitmq")

	ch, err := conn.Channel()
	failOnError(err, "failed to get a channel")

	q, err := ch.QueueDeclare("hello", false, false, false, false, nil)
	failOnError(err, "failed to declare queue")

	msgs, err := ch.Consume(q.Name, "", true, false, false, false, nil)
	failOnError(err, "failed to register a consumer")

	forever := make(chan struct{})

	go func() {
		for d := range msgs {
			log.Printf("recieved a message: %q\n", d.Body)
		}
	}()

	log.Printf(" [*] Waiting for messages. To exit press CTRL+C")

	<-forever
}
